### PR TITLE
[#5437] feat(core): add schema event for Iceberg REST server

### PR DESCRIPTION
**Pull request by orenccl**
### What changes were proposed in this pull request?

add schema event for Iceberg REST server

### Why are the changes needed?

Fix: #5437

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

unit test

### CODE REVIEW

**Comment by orenccl** [comment]
Hi, @FANNG1 

### Couple of Questions

1. Should I use `OperationType.*_SCHEMA`, or should I create a new type called `OperationType.*_NAMESPACE`?

2. Currently, there is no `SCHEMA_EXIST` or `REGISTER_TABLE`. Should I create them in this PR?

3. What should the name be for `REGISTER_TABLE`? Should it be `SCHEMA_REGISTER_TABLE`, `REGISTER_TABLE`, or something else?

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/dispatcher/IcebergNamespaceEventDispatcher.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+import org.apache.iceberg.rest.responses.UpdateNamespacePropertiesResponse;
+
+/**
+ * {@code IcebergNamespaceEventDispatcher} is a decorator for {@link
+ * IcebergNamespaceOperationExecutor} that not only delegates namespace operations to the underlying
+ * dispatcher but also dispatches corresponding events to an {@link EventBus}.
+ */
+public class IcebergNamespaceEventDispatcher implements IcebergNamespaceOperationDispatcher {
+
+  private IcebergNamespaceOperationDispatcher icebergNamespaceOperationDispatcher;
```
[the remark this answers is not in the document]
Comment (orenccl) [reply iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/dispatcher/IcebergNamespaceEventDispatcher.java]: I was following the code style from another script, but yes, the variables in this script are too long, so I will shorten them.

**Comment by FANNG1** [comment]
> 1. Should I use `OperationType.*_SCHEMA`, or should I create a new type called `OperationType.*_NAMESPACE`?

I prefer to reuse `OperationType.*_SCHEMA`, because they are the same thing with different names.

> 2. Currently, there is no `SCHEMA_EXIST` or `REGISTER_TABLE`. Should I create them in this PR?

Yes, please create it


> 3. What should the name be for `REGISTER_TABLE`? Should it be `SCHEMA_REGISTER_TABLE`, `REGISTER_TABLE`, or something else?

`REGISTER_TABLE` seems enough.

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/dispatcher/IcebergNamespaceEventDispatcher.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (FANNG1) [reply iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/dispatcher/IcebergNamespaceEventDispatcher.java]: `dispatcher` seems a better name

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/listener/api/event/IcebergRegisterTableEvent.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.annotation.DeveloperApi;
+import org.apache.gravitino.iceberg.service.IcebergRestUtils;
+import org.apache.iceberg.rest.requests.RegisterTableRequest;
+import org.apache.iceberg.rest.responses.LoadTableResponse;
+
+/** Represent an event after registering Iceberg table successfully. */
+@DeveloperApi
+public class IcebergRegisterTableEvent extends IcebergNamespaceEvent {
```
Comment (FANNG1) [inline iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/listener/api/event/IcebergRegisterTableEvent.java:30]: IcebergRegisterTableEvent is more like a table event than a namespace event though the register operation is under the namespace URL path. WDYT?

**Comment by FANNG1** [comment]
LGTM, except minor comments

**Comment by FANNG1** [comment]
@orenccl thanks for your work!

**Pull request merged**